### PR TITLE
Add tests for variable product

### LINKED ISSUE #39886: [E2E test coverage]: create Variations
Add E2E tests that validate various aspects of the variation creation process.

The tests will specifically cover the following scenarios:

- Testing the feasibility of creating variations.
- Testing the capability to edit an attribute associated with a variation.
- Testing the functionality of hiding a variation.
- Testing the ability to delete a variation.
- Testing the ability to delete an attribute.

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds E2E tests for variable products. We test the creation of a few variations and publish a new product.

Closes #39886.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run our E2E tests. All the tests should pass.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-variable-product-block-editor.spec.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+				)
+				.last()
+				.fill( productData.summary );
+
+			try {
+				await page
+					.getByLabel( 'Close Tour' )
+					.click( { timeout: 5000 } );
+			} catch ( e ) {
+				console.log( 'Tour was not visible, skipping.' );
```
Comment (octaedro) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-variable-product-block-editor.spec.js]: The tour won't be visible after closing it. That will be the case when we run the E2E tests locally more than once.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-variable-product-block-editor.spec.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+				await page
+					.getByLabel( 'Close Tour' )
+					.click( { timeout: 5000 } );
+				await page.waitForResponse(
+					( response ) =>
+						response.url().includes( '/users/1?_locale=user' ) &&
+						response.status() === 200
+				);
+			} catch ( e ) {
+				console.log( 'Tour was not visible, skipping.' );
```
Comment (octaedro) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-variable-product-block-editor.spec.js]: Same as above. We will only close the tour when it's visible.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-variable-product-block-editor.spec.js (hunk 1 above)
Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-variable-product-block-editor.spec.js]: You should not need to close the tour, it's already disabled in the `page` fixture from `./block-editor-fixtures`.

Comment (octaedro) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-variable-product-block-editor.spec.js]: Good call @adimoldovan. I just found [the PR](https://github.com/woocommerce/woocommerce/pull/44699) that does that. It was merged when I started to work on mine 😄
I will add to the `block-editor-fixtures` the code to disable the variation's tour too.

**Comment by adimoldovan** [comment]
Thanks @octaedro for working on these tests. While the spec tests well, I would argue against two things:

- the tests are dependent on each other. The second and the third tests depend on a product being created by the first one.  Ideally, each test should create it's own test data.
- there is no cleanup. Ideally, each test should clean up the data created.

**Comment by octaedro** [comment]
Thank you @adimoldovan for reviewing this PR.
I just addressed the changes you mentioned. Could you take another look at the PR?

**Review by adimoldovan — APPROVED** [review]
Looks good! 👍🏼

**Pull request merged**